fhevm/tee_pad_gas.go

```diff
@@ -0,0 +1,31 @@
+package fhevm
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func teePadGasRequiredGas(environment EVMEnvironment, suppliedGas uint64, input []byte) uint64 {
+
+	input = input[:minInt(32, len(input))]
+
+	gasLimit := environment.GasLimit()
+	consumedGas := gasLimit - suppliedGas
```

Comment:
@TheoXD Does this look correct to you, to get the amount of gas consumed up until the point the precompile was called?